Pod/Classes/ResizingTextView.swift

```diff
@@ -86,29 +132,58 @@ open class ResizingTextView: UITextView {
         NotificationCenter.default.removeObserver(self)
     }
     
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+        didUpdateText(self)
+    }
+    
     /*
      *  MARK: - Auto Resizing
      */
     
     @objc func didUpdateText(_ sender: AnyObject) {
-        let newHeight = heightForCurrentText()
+        var newHeight = heightForCurrentText()
+        
+        if let maxHeight = maxHeight, newHeight >= maxHeight {
+            newHeight = maxHeight
+            isMaxHeight = true
+        } else {
+            if let minHeight = minHeight, newHeight < minHeight {
+                newHeight = minHeight
+            }
+            isMaxHeight = false
+        }
+        
         if newHeight != heightConstraint.constant {
             updateHeightConstraint(newHeight)
         }
     }
     
     private func updateHeightConstraint(_ newHeight: CGFloat) {
         willChangeHeight?(newHeight)
-        heightConstraint.constant = newHeight
-        setNeedsLayout()
-        
-        UIView.animate(withDuration: resizeDuration,
-                       delay: 0,
-                       options: .layoutSubviews,
-                       animations: { self.layoutIfNeeded() },
-                       completion: { [weak self] _ in
-                        self?.didChangeHeight?(newHeight)
-            })
+        UIView.transition(with: self,
+                          duration: resizeDuration,
+                          animations: { [weak self, newHeight] in
+                            self?.heightConstraint.constant = newHeight
+            },
+                          completion: { [weak self, newHeight] _ in
+                            self?.didChangeHeight?(newHeight)
+                            self?.setContentOffset(CGPoint(x: 0, y: 0), animated: true)
+        })
+    }
+    
+    private func calculateHeight(for resizableHeight: ResizableHeight) -> CGFloat? {
+        switch resizableHeight {
+        case .height(let height): return height
+        case .numberOfLines(let lines): return height(forNumberOfLines: lines)
+        case .infinite: return nil
+        }
+    }
+    
+    private func height(forNumberOfLines numberOfLines: Int) -> CGFloat? {
+        guard let font = font else { return nil }
+        let padding = contentInset.top + contentInset.bottom + textContainerInset.top + textContainerInset.bottom
+        return (font.lineHeight * CGFloat(numberOfLines)) + padding
```

Comment:
> If the `maxHeight` is being used and the font is `nil`, we should fail to handle the height restriction and just log an error to tell the developer that this is wrong. "Max height cannot be calculated when font is 'nil' for (self)".

I currently just have it handling the height as if it is set to `.infinite` when the font is `nil`.
I wasn't sure how we would want the height to be handled in this case. Were you saying you wanted it not to be resized, or were you just saying that there should be some logging involved?

Reply:
Yeah, using `.infinite` is fine. I was thinking we could add some logging to warn the developer they messed up. Not super important though.